tensorflow_probability/examples/jupyter_notebooks/Probabilistic_Matrix_Factorization_TFP.ipynb

```diff
@@ -0,0 +1,926 @@
+        "D = 3  #@param {type:\"number\"} Number of latent factors\n",
+        "\n",
+        "# Creating our synthetic data\n",
+        "U_true = evaluate(tfd.Normal(loc=0., scale=1.).sample(sample_shape=[D, N]))\n",
+        "V_true = evaluate(tfd.Normal(loc=0., scale=1.).sample(sample_shape=[D, M]))\n",
+        "print(\"Shape of our matrix of users and latent factors (U_true):\", U_true.shape)\n",
+        "print(\"Shape of our matrix of videos and latent factors (V_true):\", V_true.shape)"
+      ],
+      "execution_count": 0,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Shape of our matrix of users and latent factors (U_true): (3, 50)\n",
+            "Shape of our matrix of videos and latent factors (V_true): (3, 60)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "metadata": {
+        "id": "8uoyXJVRRs4Y",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Data\n",
+        "\n",
+        "Here, we're creating our Training and Test Data. our `build_toy_dataset` function takes in the matrix of the $\\text{Users} \\times \\text{latent dimensions}$, the $\\text{Videos} \\times \\text{latent dimensions}$, the numbers of users and videos, and a parameter for noise (it's a toy dataset, but we'd at least like to add _some_ unpredictability to it). This returns Our $N \\times M$ matrix of our users and their movies of choice.\n",
+        "\n",
+        "<img src=\"https://camo.githubusercontent.com/ddd2be377ea2598056c581fb4d4b152da6ae1970/68747470733a2f2f63646e2d696d616765732d312e6d656469756d2e636f6d2f6d61782f313736302f312a32692d474a4f374a5830597a363439386a55766845672e706e67\" alt=\"Girl in a jacket\" width=\"600\">"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "NpjUesaxRs4Z",
+        "colab_type": "code",
+        "outputId": "df483aaa-9093-470d-edc0-b9713b8e80cd",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 381
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "def build_toy_dataset(U, V, N, M, noise_std=0.1):\n",
+        "    \"\"\"\n",
+        "    Function for building our toy dataset.\n",
+        "        \n",
+        "    Args:\n",
+        "      U: Integer\n",
+        "      V: Integer\n",
+        "      N: Integer Number of Users\n",
+        "      M: Integer Number of Movies\n",
+        "      noise_std: a float used for determining the standard deviation of our Normal\n",
+        "        sampling distribution. The SD is effetively equivalent to the amount of noise.\n",
+        "    Returns: \n",
+        "      np.array-like dataframe representing our data of choice for users and their\n",
+        "        movies of choice.\n",
+        "    \"\"\"  \n",
+        "    #R = np.dot(np.transpose(U), V) + np.random.normal(0, noise_std, size=(N, M))\n",
```

Comment:
Would recommend that all floats have a trailing . (e.g. np.random.normal(0., ...). Here and elsewhere.